### PR TITLE
Add issue templates, keyword labeler and PR labeler to cookiecutter

### LINKED ISSUE #3: Add issue templates
* Add issue templates. Example: https://github.com/frequenz-floss/frequenz-sdk-python/tree/v0.x.x/.github/ISSUE_TEMPLATE
* Make sure to update `bug.yml`
  * The `part` field options should match all the `part:xxx` labels defined in the repo
* Make sure to update `config.yml`:
  * Update the links if copying and pasting the templates from another repo

### LINKED ISSUE #4: Add keyword labeler app
Add [https://github.com/ZeWaka/KeywordLabeler] app configuration in `.github/keylabeler.yml`.

For example:

```yml
# KeywordLabeler app configuration. For more information check:
# https://github.com/ZeWaka/KeywordLabeler#readme

# Determines if we search the title (optional). Defaults to true.
matchTitle: true

# Determines if we search the body (optional). Defaults to true.
matchBody: true

# Determines if label matching is case sensitive (optional). Defaults to true.
caseSensitive: true

# Explicit keyword mappings to labels. Form of match:label. Required.
labelMappings:
  "part:docs": "part:docs"
  "part:tests": "part:tests"
  "part:tooling": "part:tooling"
  "part:❓": "part:❓"
```

* Make sure all the `part:xxx` labels defined in the repo are present.
* Ideally sort the `labelMappings`

SDK example: https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/.github/keylabeler.yml

### LINKED ISSUE #5: Add PR Labeler
### Workflow

Add a `.github/workflow/labeler.yml` workflow:

```yml
name: Pull Request Labeler

# XXX: !!! SECURITY WARNING !!!
# pull_request_target has write access to the repo, and can read secrets. We
# need to audit any external actions executed in this workflow and make sure no
# checked out code is run (not even installing dependencies, as installing
# dependencies usually can execute pre/post-install scripts). We should also
# only use hashes to pick the action to execute (instead of tags or branches).
# For more details read:
# https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
on: [pull_request_target]

jobs:
  Label:
    permissions:
      contents: read
      pull-requests: write
    runs-on: ubuntu-latest
    steps:
      - name: Labeler
        # Only use hashes, see the security comment above
        uses: actions/labeler@e54e5b338fbd6e6cdb5d60f51c22335fc57c401e  # 4.0.1
        with:
          repo-token: "${{ secrets.GITHUB_TOKEN }}"
```

### Configuration

Configure the workflow creating a `.github/labeler.yml`.

For example:

```yml
# Configuration for the Labeler GitHub action, executed by
# .github/workflows/labeler.yml.
#
# The basic syntax is [label]: [path patterns].
#
# For more details on the configuration please see:
# https://github.com/marketplace/actions/labeler

"part:docs": 
  - "**/*.md"
  - LICENSE

"part:tests":
  - "tests/**"

"part:tooling":
  - ".git*"
  - ".git*/**"
  - "**/*.toml"
  - "**/*.ini"
  - CODEOWNERS
  - MANIFEST.in
  - "*requirements*.txt"
  - setup.py
  - setup.cfg

"part:BLAH":
  - "src/frequenz/BLAH/**"
```

* Ideally make sure all the `part:xxx` labels defined in the repo are present.

SDK Example:
* https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/.github/labeler.yml
* https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/.github/workflows/labeler.yml

### PR DESCRIPTION
**Pull request by llucax**
Fixes #3, fixes #4, fixes #5.

### CODE REVIEW

**Comment by llucax** [comment]
Converted to draft because I think it makes more sense to combine with #4 and #5 too (because all depend heavily on labels)

**Comment by llucax** [comment]
Ready again.

File: cookiecutter/{{cookiecutter.github_repo_name}}/.github/ISSUE_TEMPLATE/bug.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        apply.
+      multiple: true
+      options:
+        - I don't know (part:❓)
+        - Documentation (part:docs)
+        - Unit, integration and performance tests (part:tests)
+        - Build script, CI, dependencies, etc. (part:tooling)
+        # TODO(cookiecutter): Add other parts
+        # Please have in mind that that the part:xxx should be created as
+        # a issue label in the GitHub repository"
```
Comment (shsms) [inline cookiecutter/{{cookiecutter.github_repo_name}}/.github/ISSUE_TEMPLATE/bug.yml:55]: should be -> need to be, to make it sound less like a suggestion and more like a requirement.

```suggestion
        # Please have in mind that that the part:xxx labels need to
        # be created in the GitHub repository.

File: cookiecutter/{{cookiecutter.github_repo_name}}/.github/keylabeler.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# Explicit keyword mappings to labels. Form of match:label. Required.
+labelMappings:
+  "part:docs": "part:docs"
+  "part:tests": "part:tests"
+  "part:tooling": "part:tooling"
+  "part:❓": "part:❓"
+  # TODO(cookiecutter): Add other parts
+  # Please have in mind that that the part:xxx should be created as a issue
+  # label in the GitHub repository"
```
Comment (shsms) [inline cookiecutter/{{cookiecutter.github_repo_name}}/.github/keylabeler.yml:21]: same as ⬆️

File: cookiecutter/{{cookiecutter.github_repo_name}}/.github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# For example:
+#
+# "part:module":
+#   - "src/frequenz/{{cookiecutter.type}}/{{cookiecutter.name}}/module/**"
+#
+# "part:other":
+#   - "src/frequenz/{{cookiecutter.type}}/{{cookiecutter.name}}/other/**"
+#
+# Please have in mind that that the part:xxx should be created as a issue label
+# in the GitHub repository"
```
Comment (shsms) [inline cookiecutter/{{cookiecutter.github_repo_name}}/.github/labeler.yml:19]: ➕

**Review by shsms — APPROVED** [review]
LGTM except for one optional comment about a comment in 3 places.

**Comment by llucax** [comment]
Maybe I should have mentioned that auto-merge was enabled. I think from now on any PR I open and it's not a draft will have auto-merge enabled.

**Comment by llucax** [comment]
Will fix in a different PR.

**Comment by llucax** [comment]
Personally if I have comments I don't approve if auto-merge is enabled to give the PR author the opportunity to make adjustments, but as you say, if it is only optional it is also fine if you prefer to just approve to keep things moving :)

**Pull request merged**